api/v1alpha1/proxydefaults_types.go

```diff
@@ -54,6 +54,8 @@ type ProxyDefaultsSpec struct {
 	// Config is an arbitrary map of configuration values used by Connect proxies.
 	// Any values that your proxy allows can be configured globally here.
 	// Supports JSON config values. See https://www.consul.io/docs/connect/proxies/envoy#configuration-formatting
+	// +kubebuilder:validation:Type=object
+	// +kubebuilder:validation:Schemaless
```

Comment:
This is due to https://github.com/kubernetes-sigs/controller-tools/pull/528#pullrequestreview-577871646 getting merged and now released in latest controller-tools version. It now supports `object` kind without breaking codegen